src/controller/testController.ts

```diff
@@ -265,23 +265,47 @@ function enqueueTestMethods(testItems: TestItem[], run: TestRun): void {
  * @returns
  */
 async function getIncludedItems(request: TestRunRequest, token?: CancellationToken): Promise<TestItem[]> {
-   let testItems: TestItem[] = [];
-   if (request.include) {
-       testItems.push(...request.include);
-   } else {
-       testController?.items.forEach((item: TestItem) => {
-           testItems.push(item);
-       });
-   }
-   if (testItems.length === 0) {
-       return [];
-   }
-   removeTestInvocations(testItems);
-   testItems = await expandTests(testItems, TestLevel.Class, token);
-   // @ts-expect-error
-   const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
-   testItems = _.differenceBy(testItems, excludingItems, 'id');
-   return testItems;
+    let testItems: TestItem[] = [];
+    if (request.include) {
+        testItems.push(...request.include);
+    } else {
+        testController?.items.forEach((item: TestItem) => {
+            testItems.push(item);
+        });
+    }
+    if (testItems.length === 0) {
+        return [];
+    }
+    handleInvocationRerun(testItems);
+    removeTestInvocations(testItems);
+    testItems = await expandTests(testItems, TestLevel.Class, token);
+    // @ts-expect-error
+    const excludingItems: TestItem[] = await expandTests(request.exclude || [], TestLevel.Class, token);
+    testItems = _.differenceBy(testItems, excludingItems, 'id');
+    return testItems;
+}
+
+function handleInvocationRerun(testItems: TestItem[]): void {
+
+    // always remove uniqueIds from non-invocation items, since they would have been set for a past run
+    testItems.forEach((item: TestItem) => {
+        if (dataCache.get(item) && dataCache.get(item)!.testLevel !== TestLevel.Invocation) {
```

Comment:
How about extract `dataCache.get(item)` to a local variable? That would make the code here more neat.